src/SFGUI/UI/XMLLoader.cpp

```diff
@@ -0,0 +1,517 @@
+#include <SFGUI/UI/XMLLoader.hpp>
+#include <algorithm>
+#include <string>
+#include <unistr.h>
+#include <uniconv.h>
+#include <dlfcn.h>
+
+// pass string to lowercase
+static std::string toLowercase(std::string str) {
+    std::transform(str.begin(), str.end(), str.begin(),
+                   [](unsigned char c) { return std::tolower(c); });
+    return str;
+}
+
+// this function is used because sf::String can't be parse special characters from string to string32
+static std::u32string utf8_to_utf32(std::string utf8_str) {
+    std::u32string result;
+
+    size_t size;
+    uint32_t* utf32 = u8_to_u32(reinterpret_cast<const uint8_t*>(utf8_str.data()), utf8_str.size(), nullptr, &size);
+    if(!utf32) return U"";
+
+    result.assign(utf32, utf32 + size);
+    free(utf32);
+    return result;
+}
```

Comment:
I don't think we need a whole new library to parse strings.

SFML's `sf::Utf<T>` utilities can certainly convert from UTF8 to UTF32